src/python/pants/backend/java/compile/javac_subsystem.py

```diff
@@ -23,5 +23,8 @@ def register_options(cls, register):
             advanced=True,
             help="The JDK to use for invoking javac."
             " This string will be passed directly to Coursier's `--jvm` parameter."
-            " Run `cs java --available` to see a list of available JVM versions on your platform.",
+            " Run `cs java --available` to see a list of available JVM versions on your platform."
+            " If the string 'system' is passed, Coursier's `--system-jvm` option will be used"
+            " instead, but note that this can lead to inconsistent behavior since the JVM version"
+            " will be whatever happens to be found first on the system's PATH.",
```

Comment:
Assuming that the above safety improvement is implemented, perhaps rather than replacing the `--jdk` option, the "system" option should be next to it: something like `--jdk-use-system`.

The reasoning is that if we are already querying coursier to determine which JDK it located on the system, we could also enforce a particular `--jdk` setting (i.e. `--jdk=14 --jdk-use-system` would confirm that `coursier` found `14` already installed, and fail otherwise). That would also have the advantage of allowing the `--jdk-use-system` flag to continue to be useful even when individual targets are setting their JDK versions and the global `--jdk` value evolves into being just a default.

Reply:
I mulled this for a bit, and while I think it's reasonable, I can't think of a good way to consistently parse out the JDK version.  There isn't an obvious reverse mapping between the output of `java -version` and Coursier's `--jvm` selectors.

Given that JVM selection options are still very much in flux in the current JVM backends, I propose to punt this problem to later when we have a better idea of how target-level JDK constraints are going to interact with subsystem and rule level JVM selection.  If we add an explicit `jdk-use-system` later, it's easy to make `jdk=system` an alias for that without breaking any workflows; whereas splitting the option out now starts to constrain our interface in ways we might want to change in the near future.

What do you think?

Reply:
Yep, totally fine!